### PR TITLE
LKMM frontend

### PR DESCRIPTION
**Pull request by hernanponcedeleon**
This PR includes almost every change that was done in the `cna-verification` branch, but keeps the architecture precision to `-1` (as I still haven't find a clean way to make this a proper configuration option).

### CODE REVIEW

**Comment by hernanponcedeleon** [comment]
Some of the changes in `ConstantPropagation` from the `cna-verification` branch introduce a bug. For this PR I kept the version of CP from `development` and [added a note](https://github.com/hernanponcedeleon/Dat3M/blob/70904b1975f8cfbb3fdf4d1b8618be6b779e86ba/dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ConstantPropagation.java#L190-L192) to `cna-verification` about the problem.

File: cat/lkmm-fix.cat [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ *)
+
+"Linux-kernel memory consistency model"
+
+(*
+ * File "lock.cat" handles locks and is experimental.
+ * It can be replaced by include "cos.cat" for tests that do not use locks.
+ *)
+
+include "lock.cat"
```
Comment (ThomasHaas) [inline cat/lkmm-fix.cat:21]: This is not true for the "fixed" version, is it?

File: dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/ProgramParser.java [hunk 2]
```diff
@@ -36,18 +37,22 @@ public Program parse(File file) throws Exception {
...
+    public Program parse(String raw, String path, String format, String cflags) throws Exception {
         switch (format) {
         	case "c":
         	case "i":
-				File CFile = File.createTempFile("dat3m", ".c");
+				File CFile = path.equals("") ?
+						// This if for the case where the user fully type the program instead of loading it
+						File.createTempFile("dat3m", ".c") :
+						// This if for the case where the user loaded the program 						
+						new File(path, "dat3m.c");
```
Comment (ThomasHaas) [inline dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/ProgramParser.java:48]: Typos in comments.

File: dartagnan/src/main/resources/log4j2.xml [hunk 3]
```diff
@@ -4,25 +4,25 @@
...
     <File name="Refinement_File" fileName="output/logs/refinement.log">
-      <PatternLayout pattern="%d{HH:mm:ss} [%level] %C{1}.%M - %msg%n"/>
+      <PatternLayout pattern="%d{[dd.MM.yyyy] HH:mm:ss} [%level] %C{1}.%M - %msg%n"/>
     </File>
     
   </Appenders>
   
   <Loggers>
-    <Logger name="com.dat3m.dartagnan.verification.solving.RefinementSolver" level="info" additivity="false">
+    <Logger name="com.dat3m.dartagnan.verification.solving.RefinementSolver" level="debug" additivity="false">
```
Comment (ThomasHaas) [inline dartagnan/src/main/resources/log4j2.xml:25]: We want Debug as default now?

File: cat/lkmm-fix.cat (hunk 1 above)
Comment (hernanponcedeleon) [reply cat/lkmm-fix.cat:21]: The fixed versions were not intended to be part of this PR.

File: dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/ProgramParser.java (hunk 2 above)
Comment (hernanponcedeleon) [reply dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/ProgramParser.java:48]: "fully typed"?

File: dartagnan/src/main/resources/log4j2.xml (hunk 3 above)
Comment (hernanponcedeleon) [reply dartagnan/src/main/resources/log4j2.xml:25]: No

File: dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/ProgramParser.java (hunk 2 above)
Comment (ThomasHaas) [reply dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/ProgramParser.java:48]: That one, but also ´if´->`is`

**Pull request merged**